stackslib/src/chainstate/stacks/transaction.rs

```diff
@@ -686,19 +689,24 @@ impl StacksTransaction {
                 )));
             }
         };
+        let tx = StacksTransaction {
+            version,
+            chain_id,
+            auth,
+            anchor_mode,
+            post_condition_mode,
+            post_conditions,
+            payload,
+        };
 
-        Ok((
-            StacksTransaction {
-                version,
-                chain_id,
-                auth,
-                anchor_mode,
-                post_condition_mode,
-                post_conditions,
-                payload,
-            },
-            fd.num_read(),
-        ))
+        if !StacksBlock::validate_transactions_static_epoch(&[tx.clone()], epoch_id, false) {
+            warn!("Invalid tx: target epoch is not activated");
```

Comment:
I'd drop this entirely.  It should be fine to decode a transaction regardless of what epoch we're in.  What ultimately matters is that we don't process a transaction that isn't supported in the current epoch.  To support that, we would also treat a block as invalid if it contained an unsupported transaction (so we don't even store it or relay it).